File: DDD5/ems/src/main/java/com/mt/cms/service/newsComment/NewsCommentService.java
package com.mt.cms.service.newsComment;

import com.mt.common.core.web.base.PageDTO;
import com.mt.common.core.web.base.PageResultDTO;
import com.mt.cms.entity.newsComment.NewsComment;

import java.util.List;

public interface NewsCommentService {
    /**
     * 根据分页参数查询新闻评论集合
     *
     * @param pageDTO 分页条件
     */
    public PageResultDTO findNewsComments(PageDTO pageDTO);

    /**
     * 查询全部新闻评论集合
     *
     */
    public List<NewsComment> findAllNewsComments();

    /**
     * 根据名称查询新闻评论集合(只提取ID 和 Name)
     *
     * @param newsCommentName 名称
     */
    public List<NewsComment> findNewsCommentsWithIdNameByName(String newsCommentName);

    /**
     * 查询所有新闻评论集合(只提取ID 和 Name)
     *
     */
    public List<NewsComment> findAllNewsCommentsWithIdName();

    /**
     * 根据ID查询指定的新闻评论(只提取ID 和 Name)
     *
     * @param newsCommentId Id
     */
    public NewsComment findNewsCommentsWithIdNameById(Long newsCommentId);

    /**
     * 根据ID查询指定的新闻评论
     *
     * @param newsCommentId Id
     */
    public NewsComment findNewsComment(Long newsCommentId);

    /**
     * 根据ID查询指定的新闻评论(包含外键)
     *
     * @param newsCommentId Id
     */
    public NewsComment findNewsCommentWithForeignName(Long newsCommentId);

    /**
     * 新增新闻评论
     *
     * @param newsComment 实体对象
     */
    public NewsComment saveNewsComment(NewsComment newsComment);

    /**
     * 更新新闻评论
     *
     * @param newsComment 实体对象
     */
    public NewsComment updateNewsComment(NewsComment newsComment);

    /**
     * 根据ID删除新闻评论
     *
     * @param newsCommentId ID
     */
    public void deleteNewsComment(Long newsCommentId);
}

File: DDD5_UI/src/views/cms/newsComment/newsComment/newsCommentService.js

import axios from 'axios';

/**
 * 新闻评论
 */
export class NewsCommentService {
    // 分页查询列表
    static findNewsComments (queryParams) {
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsComments',
            method: 'POST',
            data: queryParams
        });
    }

    //根据ID查询指定的新闻评论
    static findNewsComment (newsCommentId) {
        const query = {
            newsCommentId
        }
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsComment',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的新闻评论(包含外键名称)
    static findNewsCommentForView (newsCommentId) {
        const query = {
            newsCommentId
        }
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsCommentForView',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的新闻评论(包含外键名称)
    static findNewsCommentForEdit (newsCommentId) {
        const query = {
            newsCommentId
        }
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsCommentForEdit',
            method: 'POST',
            params: query,
        });
    }

    //根据ID查询指定的新闻评论(只提取ID 和 Name)
    static findNewsCommentsWithIdNameById (newsCommentId) {
        const query = {
            newsCommentId
        }
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsCommentsWithIdNameById',
            method: 'POST',
            params: query,
        });
    }

    //根据名称查询新闻评论集合(只提取ID 和 Name)
    static findNewsCommentsWithIdNameByName (newsCommentName) {
        const query = {
            newsCommentName
        }
        return axios({
            url: '/cms/newsComment/NewsComment/findNewsCommentsWithIdNameByName',
            method: 'POST',
            params: query,
        });
    }

    //新增保存新闻评论
    static saveNewsComment (newsComment) {
        const query = {}
        return axios({
            url: '/cms/newsComment/NewsComment/saveNewsComment',
            method: 'POST',
            params: query,
            data: newsComment
        });
    }

    //修改保存新闻评论
    static updateNewsComment (newsComment) {
        const query = {}
        return axios({
            url: '/cms/newsComment/NewsComment/updateNewsComment',
            method: 'POST',
            params: query,
            data: newsComment
        });
    }

    //根据ID查询指定的新闻评论(只提取ID 和 Name)
    static deleteNewsComment (newsCommentId) {
        const query = {
            newsCommentId
        }
        return axios({
            url: '/cms/newsComment/NewsComment/deleteNewsComment',
            method: 'POST',
            params: query,
        });
    }
}
